### PR TITLE
CAT-FIX Several Bugs and UX Issues in Scenes, PenBricks and Bubbles:

### PR DESCRIPTION
**Pull request by loibi93**
- fixed all bugs with back Button
- Merge Scene Dialog -> Selection is now Scene 1 and Scene 2 by default
- Fix missing show details after project deletion
- Fix Play Dialog in SceneListFragment
- Fix Title bar in Object Menu
- Fixed Brick Category

### CODE REVIEW

**Comment by p4p4** [comment]
ok to test

**Comment by p4p4** [comment]
retest this please

**Comment by loibi93** [comment]
I can't because I am not in the catrobat member list ;)

**Comment by p4p4** [comment]
obviously I cant start it either for you (even though I have the label "Catrobat member") but I thought that I could give it a try ;-)

**Comment by loibi93** [comment]
@thmq would you be so kind :D

**Comment by p4p4** [comment]
Instead, you could simply run [static check](https://jenkins.catrob.at/view/LeeroyCatroid/job/CatroidCheck/) and [single package test](https://jenkins.catrob.at/job/CatroidSinglePackageEmulatorTest/) on your own, and optionally post the URLs to the results in the description. That should provide extra confidence to your high quality code. ;)

**Comment by robertpainsi** [comment]
retest this please

**Comment by robertpainsi** [comment]
ok to test

**Comment by robertpainsi** [comment]
Hmm, interesting... Maybe it currently doesn't work?
(Removing comments to keep the PR clean)

**Comment by loibi93** [comment]
Here is the static check: https://jenkins.catrob.at/view/LeeroyCatroid/job/CatroidCheck/353/
Single Package test seems to be broken ;) @thmq

File: catroid/src/main/java/org/catrobat/catroid/ui/ProgramMenuActivity.java [hunk 1]
```diff
@@ -82,7 +82,7 @@ public void onCreate(Bundle savedInstanceState) {
 			Sprite sprite = ProjectManager.getInstance().getCurrentSprite();
 			Scene scene = ProjectManager.getInstance().getCurrentScene();
 			if (sprite != null && scene != null && actionBar != null) {
-				String title = scene.getName() + ": " + sprite.getName();
+				String title = ProjectManager.getInstance().getCurrentProject().getSceneList().size() > 1 ? scene.getName() + ": " + sprite.getName() : sprite.getName();
```
Comment (robertpainsi) [inline catroid/src/main/java/org/catrobat/catroid/ui/ProgramMenuActivity.java]: * Do not use concatenation for strings which are visible to the user.
* You should create a helper method to create the sprite title.

File: catroid/src/main/java/org/catrobat/catroid/ui/fragment/ScenesListFragment.java [hunk 2]
```diff
@@ -383,7 +384,8 @@ public void onClick(DialogInterface dialog, int id) {
 	private void checkSceneCountAfterDeletion() {
 		ProjectManager projectManager = ProjectManager.getInstance();
 		if (projectManager.getCurrentProject().getSceneList().size() == 0) {
-			Scene emptyScene = new Scene(getActivity(), String.format(getString(R.string.default_scene_name), 1), projectManager.getCurrentProject());
+			Scene emptyScene = new Scene(getActivity(), String.format(getString(R.string.default_scene_name), Integer.toString(1)), projectManager
+					.getCurrentProject());
```
Comment (robertpainsi) [inline catroid/src/main/java/org/catrobat/catroid/ui/fragment/ScenesListFragment.java]: Is this change really necessary?

File: catroid/src/main/java/org/catrobat/catroid/ui/ProgramMenuActivity.java (hunk 1 above)
Comment (loibi93) [inline catroid/src/main/java/org/catrobat/catroid/ui/ProgramMenuActivity.java]: Why ?

File: catroid/src/main/java/org/catrobat/catroid/formulaeditor/SensorHandler.java [hunk 3]
```diff
@@ -595,12 +595,12 @@ public void onLocationChanged(Location location) {
...
 			longitude = location.getLongitude();
 			locationAccuracy = location.getAccuracy();
-			altitude = location.hasAltitude() ? location.getAltitude() : Double.NaN;
+			altitude = location.hasAltitude() ? location.getAltitude() : 0;
 		} else if (!isGpsConnected) {
 			latitude = location.getLatitude();
 			longitude = location.getLongitude();
 			locationAccuracy = location.getAccuracy();
-			altitude = location.hasAltitude() ? location.getAltitude() : Double.NaN;
+			altitude = location.hasAltitude() ? location.getAltitude() : 0;
```
Comment (loibi93) [inline catroid/src/main/java/org/catrobat/catroid/formulaeditor/SensorHandler.java]: Fixed.

File: catroid/src/main/java/org/catrobat/catroid/ui/fragment/ScenesListFragment.java (hunk 2 above)
Comment (loibi93) [inline catroid/src/main/java/org/catrobat/catroid/ui/fragment/ScenesListFragment.java]: Fixed.

File: catroid/src/main/java/org/catrobat/catroid/ui/ProgramMenuActivity.java (hunk 1 above)
Comment (robertpainsi) [inline catroid/src/main/java/org/catrobat/catroid/ui/ProgramMenuActivity.java]: Internationalization, RTL support. However, thinking about it, it's not that easy after all.

I heard from @wslany that someone is (or was) working on RTL support. What's the progress on that? What does he suggests?

Comment (bernadettespieler) [inline catroid/src/main/java/org/catrobat/catroid/ui/ProgramMenuActivity.java]: @AyyalAwwad is concerning with RTL support but this will probably still last some time

Comment (AyyalAwwad) [inline catroid/src/main/java/org/catrobat/catroid/ui/ProgramMenuActivity.java]: All your comments and questions are welcome.
Different issues in software localization to BiDi-language should be considered, my progress is very  good, and  I have faced many correlated challenging aspects for localization to these languages. The characteristics of BiDi-languages can raise issues in translation. many thanks.
